executor/grant.go

```diff
@@ -163,12 +168,19 @@ func (e *GrantExec) Next(ctx context.Context, req *chunk.Chunk) error {
 				return err
 			}
 		}
+
+		// Previously "WITH GRANT OPTION" implied setting the Grant_Priv in mysql.user.
+		// However, with DYNAMIC privileges the GRANT OPTION is individually grantable, and not a global
+		// property of the user. The logic observed in MySQL 8.0 is as follows:
+		// - The GRANT OPTION applies to all PrivElems in e.Privs.
+		// - Thus, if PrivElems contains any non-DYNAMIC privileges, the user GRANT option needs to be set.
+		// - If it contains ONLY dynamic privileges, don't set the GRANT option, as it is individually set in the handling of dynamic options.
 		privs := e.Privs
-		if e.WithGrant {
+		if e.WithGrant && containsNonDynamicPriv(privs) {
 			privs = append(privs, &ast.PrivElem{Priv: mysql.GrantPriv})
 		}
 
-		// Grant global priv to user.
+		// Grant TLS privs to use in global table
```

Comment:
What does 'TLS privs' mean?

Reply:
There is a table called "global_privs" and another one called "global_grants", which may be confusing because a priv and a grant mean the same thing in many contexts.

"global_privs" store TLS configuration (this is TiDB specific behavior; in MySQL it is stored in the `user` table). "global_grants" stores dynamic privileges. The TiDB implementation is compatible with MySQL.

This comment attempts to reduce the confusion by reminding us that it is about the TLS settings.

Reply:
I see, thanks!